Define behaviour of executeTransaction and its return data (#4)
5afe/safe-protocol-specs: mediator/README.md

```solidity
interface ISafeProtocolMediator {
    function executeTransaction(Safe safe, SafeTransaction tx) external view returns (bytes[] memory data);
    function executeRootAccess(Safe safe, SafeRootAccess rootAccess) external view returns (bytes memory data);
}
```

To handle return data for `executeTransaction` (as it is an array of actions)

> - Size of `data` must be equal to size of actions array.

Reviewer: ```suggestion
- The size of `data` must equal the size of the actions array.
```

> - Each element in `data` corresponds to the action that has been execution i.e. `data[i]` is the result of `action[i]`.

Reviewer: ```suggestion
- Each element in `data` corresponds to the action that has been executed i.e. `data[i]` is the result of `action[i]`.
```

```solidity
interface ISafeProtocolMediator {
    function executeTransaction(Safe safe, SafeTransaction tx) external view returns (bool success, bytes[] memory data);
    function executeRootAccess(Safe safe, SafeRootAccess rootAccess) external view returns (bool success, bytes memory data);
}
```

To handle return data for `executeTransaction` (as it is an array of actions)
- Size of `data` must be equal to size of actions array.
- Each element in `data` corresponds to the action that has been execution i.e. `data[i]` is the result of `action[i]`

> - On failed execution of any one of the action, all elements of `data` must be empty

Reviewer 2: how do we return the error message?

> - TODO: Should transaction revert on failed execution of any one of the actions?

Reviewer 2: the behaviour should be the same as in the multisend library: if one fail all revert

Author: Updated implementation: [https://github.com/5afe/safe-protocol/commit/0fc1f769c2abc2e2f0bc26541d303c39a8f3debd#diff-db822b48ccd482d4ada80336[…]120adcb9b2f96324d8698a5fd37da3](https://github.com/5afe/safe-protocol/commit/0fc1f769c2abc2e2f0bc26541d303c39a8f3debd#diff-db822b48ccd482d4ada803363dfedaf141120adcb9b2f96324d8698a5fd37da3)

> - On failed execution of any one of the action, all elements of `data` must be empty.

Reviewer 2: This is not in-line with the latest proposal

Author: updated
